src/common/heap_profiling/src/jeprof.rs

```diff
@@ -14,12 +14,25 @@
 
 use std::path::Path;
 use std::process::Command;
+use std::result::Result;
 use std::{env, fs};
 
-use anyhow::anyhow;
-use risingwave_common::error::Result;
+/// Error type for running `jeprof`.
+#[derive(thiserror::Error, Debug, thiserror_ext::ContextInto)]
+pub enum JeprofError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
 
-pub async fn run(profile_path: String, collapsed_path: String) -> Result<()> {
+    #[error("jeprof exit with an error (stdout: {stdout}, stderr: {stderr}): {inner}")]
+    ExitError {
+        #[source]
+        inner: std::process::ExitStatusError,
+        stdout: String,
+        stderr: String,
```

Comment:
`#[source]` will generate `From<ExitStatusError>` for `JeprofError` right? But how are `stdout` and `stderr` set?

Does this work because of `thiserror_ext::ContextInto` which generates `into_exit_error` which will retain the source error as `inner` and set the arguments for `stdout` and `stderr`?

Reply:
> Does this work because of `thiserror_ext::ContextInto` which generates `into_exit_error` which will retain the source error as `inner` and set the arguments for `stdout` and `stderr`?

Exactly! I'll refine the documentation of `thiserror_ext` once it gets more stable.

Reply:
> `#[source]` will generate `From<ExitStatusError>` for `JeprofError` right?

This is not accurate. Only `#[from]` will generate `From` impl, while `#[source]` only helps us to correctly maintain the source chain with `source()` method.